File: rrtx/Node.py
import math, sys

class Node(object):
    def __init__(self, xcor, ycor):
        self.xcor = xcor
        self.ycor = ycor
        
        self.incoming_0 = set()
        self.outgoing_0 = set()
        
        self.incoming_r = set()
        self.outgoing_r = set()
        
        self.g = float('inf')
        self.lmc = float('inf')
        
        self.parent = None
        self.children = set()

    def getCoords(self):
        return (self.xcor, self.ycor)

    def getIntCoords(self):
        return (int(self.xcor), int(self.ycor))

    def distance(self, other):
        other_x, other_y = other.getCoords()
        return math.sqrt((self.xcor - other_x)**2 + (self.ycor - other_y)**2)

    def __lt__(self, other):
        return self

File: rrtx/Node.cpp
//
// Created by Ian Pérez on 10/26/19.
//

#include "Node.h"

Node::Node(float xc, float yc){
    xcor = xc;
    ycor = yc;

    incoming_0 = std::vector<Node*>();
    outgoing_0 = std::vector<Node*>();

    incoming_r = std::vector<Node*>();
    outgoing_r = std::vector<Node*>();

    g = std::numeric_limits<float>::infinity();
    lmc = std::numeric_limits<float>::infinity();
    parent = NULL;
    children = std::vector<Node*>();
}

Node::Node(){
    xcor = -100;
    ycor = -100;
}

std::tuple<float, float> Node::getCoords(){
    return std::make_tuple(xcor, ycor);
}

std::tuple<int, int> Node::getIntCoords(){
    return std::make_tuple((int)xcor, (int)ycor);
}

float Node::distance(Node* other){
    float other_x, other_y;
    std::tie(other_x, other_y) = other->getCoords();
    return std::sqrt(std::pow((xcor - other_x), 2) + std::pow((ycor - other_y), 2));
}

bool Node::operator==(const Node &other) const {
    return this->xcor == other.xcor && this->ycor == other.ycor;
}

bool Node::operator!=(const Node &other) const {
    return this->xcor != other.xcor || this->ycor != other.ycor;
}

bool Node::operator<(const Node &other) const {
    if(this->xcor == other.xcor){
        return this->ycor < other.ycor;
    }
    return this->xcor < other.xcor;
}

bool Node::operator>(const Node &other) const {
    return true;
}

void Node::display() {
    std::cout << std::string("\n") + "(" + std::to_string(xcor) + ", " + std::to_string(ycor) + ")\n";
}